### PR TITLE
Chore/argo workflow chart basic and monitering

### PR DESCRIPTION
**Pull request by johnku001**

### CODE REVIEW

File: cronjob/templates/_cronjob_argo_workflow.tpl [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+{{- $podGC := .Values.podGC | default dict -}}
+    workflowSpec:
+    workflowMetadata:
+      labels:
+        name: {{ .Values.name }}
+      {{- range $key, $value := .Values.annotations }}
+      {{ $key | quote }} : {{ $value | quote }}
+      {{- end }}  
+    {{- if .Values.serviceaccount }}
+    serviceAccountName: {{ .Values.serviceaccount.name | default (printf "%s-argo-workflows-admin-service-account" .Release.Namespace) }}
```
Comment (neal-sl) [inline cronjob/templates/_cronjob_argo_workflow.tpl:11]: check service account

File: cronjob/templates/_cronjob_argo_workflow.tpl [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            - key: namespace
+              value: "{{ "{{" }}workflow.namespace{{ "}}" }}"
+            - key: kind
+              value: "cron-workflow"
+          help: "Count of execution by success status"                            
+          when: "{{ "{{" }}status{{ "}}" }} == Succeededed" # Emit the metric conditionally. Works the same as normal "when"
+          counter:
+            value: "1"  # This increments the counter by 1
+    entrypoint: entry
+    {{- if or (.Values.healthCheck) (.Values.slack) (.Values.newRelic) }}
```
[the remark this answers is not in the document]
Comment (neal-sl) [reply cronjob/templates/_cronjob_argo_workflow.tpl:56]: slackApp:
  portalDomain:
  webhookUrl:

File: cronjob/templates/_argo_cron_workflow.tpl [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+            template: failed-handler
+            when: "{{ "{{" }}workflow.status{{ "}}" }} != Succeeded"
+      - name: success-handler # template for the success case
+        steps:
+        -
+          {{- if .Values.exitNotifications.slackApp }}  
+          - name: Notice-SlackApp-Succeeded
+            template: notice-slack-app-succeeded 
+          {{- end }}                          
+      - name: failed-handler  # template for the failed case
```
Comment (neal-sl) [inline cronjob/templates/_argo_cron_workflow.tpl:135]: failure-handler

File: cronjob/templates/_argo_cron_workflow.tpl [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+                name: {{ . }}
+            {{- end }}
+          {{- end }}
+      {{- if .Values.exitNotifications }}    
+      - name: exit-handler
+        steps:
+        - - name: Success
+            template: success-handler
+            when: "{{ "{{" }}workflow.status{{ "}}" }} == Succeeded" 
+          - name: Failed
```
Comment (neal-sl) [inline cronjob/templates/_argo_cron_workflow.tpl:125]: Failure

**Review by neal-sl — APPROVED** [review]

File: cronjob/templates/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+# if "exitNotifications" is not exist, then the workflow will apply default exit handle made in Infra.
+exitNotifications:
+  slackApp:
+    portalDomain: "xxxxxxxxxxxxxxxxxxxxxxxxxxx" # required if "slackApp" is exist
+    webhookUrl: "xxxxxxxxxxxxxxxxxxxxxxxxxxx"   # required if "slackApp" is exist
+  newRelic:                                     # optional, recommend pass through all the variable of "newRelic" section by pipeline
+    licenseKey: "xxxxxxxxxxx"                   # required if "newRelic" is exist
+    appName: "xxxxxxxxxxx"                      # required if "newRelic" is exist 
+  healthcheckIo:
+    successUUID: "2a3dbf43-d9b6-4992-8ac1-751dd69001cb"
```
Comment (neal-sl) [inline cronjob/templates/README.md:44]: group to uuid

File: cronjob/templates/_argo_cron_workflow.tpl [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+    {{- end }}         
+    metrics:
+      prometheus:
+        - name: cron_workflow_exec_duration_gauge # Metric name (will be prepended with "argo_workflows_")
+          labels: # Labels are optional. Avoid cardinality explosion.
+            - key: name
+              value: "{{ "{{" }}workflow.labels.name{{ "}}" }}"
+            - key: namespace
+              value: "{{ "{{" }}workflow.namespace{{ "}}" }}"              
+            - key: kind
```
Comment (neal-sl) [inline cronjob/templates/_argo_cron_workflow.tpl:28]: no need

Comment (johnku001) [reply cronjob/templates/_argo_cron_workflow.tpl]: fixed

File: cronjob/templates/README.md (hunk 5 above)
Comment (johnku001) [reply cronjob/templates/README.md]: fixed

**Review by Dose0128 — APPROVED** [review]

File: cronjob/templates/_argo_cron_workflow.tpl [hunk 7]
```diff
@@ -0,0 +1,193 @@
+{{- define "cronjob.argo_cron_workflow" -}}
+{{- $podGC := .Values.podGC | default dict -}}
```
[the remark this answers is not in the document]
Comment (johnku001) [reply cronjob/templates/_argo_cron_workflow.tpl:2]: 這樣寫error message 會比較漂亮點 ，不然會是 interface{}.xxx. not exist  比較難懂

[the remark this answers is not in the document]
Comment (johnku001) [reply cronjob/templates/_argo_cron_workflow.tpl]: fixed

**Review by acgs771126 — APPROVED** [review]

**Review by acgs771126 — APPROVED** [review]
LGTM~gogo

**Pull request merged**